Assets/SteamVR_Unity_Toolkit/Scripts/VRTK_TouchpadWalking.cs

```diff
@@ -91,16 +123,33 @@ private void FixedUpdate()
         Move();
     }
 
-    private void InitControllerListeners(GameObject controller)
+    private void SetControllerListeners(GameObject controller)
     {
-        if (controller)
+        if (controller && controller == controllerManager.left)
+        {
+            ToggleControllerListeners(controller, leftController, ref leftSubscribed);
+        }
+        else if (controller && controller == controllerManager.right)
+        {
+            ToggleControllerListeners(controller, rightController, ref rightSubscribed);
+        }
+    }
+
+    private void ToggleControllerListeners(GameObject controller, bool toggle, ref bool subscribed)
+    {
+        var controllerEvent = controller.GetComponent<VRTK_ControllerEvents>();
+        if (controllerEvent && toggle && !subscribed)
+        {
+            controllerEvent.TouchpadAxisChanged += new ControllerClickedEventHandler(DoTouchpadAxisChanged);
+            controllerEvent.TouchpadUntouched += new ControllerClickedEventHandler(DoTouchpadUntouched);
+            subscribed = true;
+        }
+        else if (controllerEvent && !toggle && subscribed)
         {
-            var controllerEvent = controller.GetComponent<VRTK_ControllerEvents>();
-            if (controllerEvent)
-            {
-                controllerEvent.TouchpadAxisChanged += new ControllerClickedEventHandler(DoTouchpadAxisChanged);
-                controllerEvent.TouchpadUntouched += new ControllerClickedEventHandler(DoTouchpadUntouched);
-            }
+            controllerEvent.TouchpadAxisChanged -= new ControllerClickedEventHandler(DoTouchpadAxisChanged);
```

Comment:
I have a feeling this can unregister the last listener on that event (even if it's not the one you're expecting).

I (think) the solution is to store your listener you're adding to a variable then subtract that stored listener to prevent this, I should probably read documentation more :p